Make things a little more robust to random files

rgetz opened the pull request:
In the past, when consuming random files, eepdump would likely segfault.
These changes make it a little more robust, by ensuring that the content
lengths are within the file length, as well as not parsing files which
are not the correct version or have the "R-Pi" signature.

Signed-off-by: Robin Getz <robin.getz@analog.com>

pelwell commented on eepromutils/eepdump.c:
```diff
@@ -31,14 +41,25 @@ int read_bin(char *in, char *outf) {
 	if (FORMAT_VERSION!=header.ver) fprintf(out, "# WARNING: format version mismatch!!!\n");
-	
-	fprintf(out, "# --Header--\n# signature=0x%08x\n# version=0x%02x\n# reserved=%u\n# numatoms=%u\n# eeplen=%u\n# ----------\n\n\n", header.signature, header.ver, header.res, header.numatoms, header.eeplen);
+	if (HEADER_SIGN!=header.signature) fprintf(out, "# WARNING: format signature mismatch!!!\n");
```
Spaces around `!=` please.

rgetz replied:
They aren't there in any if the existing code - and it's worse than a bad coding style is to have to have multiple.

pelwell commented on eepromutils/eepdump.c:
```diff
@@ -51,8 +72,14 @@ int read_bin(char *in, char *outf) {
+			printf("size of atom[%i] = %i longer than rest of file (%i)\n", i, atom.dlen, size_in);
+			goto err;
+		}
+		size_in -= atom.dlen + ATOM_SIZE + 4;
```
Is this size tracking accurate? More importantly, is it necessary? Since we know the file position at this point, a simple `pos + atom.dlen < size_in` (where size_in isn't updated) would suffice to stop us allocating insane amounts of memory for a broken file.

rgetz replied:
yes - it's accurate, it's not much different than tracking the position of the file (which the existing code does), but it checks as it goes along, rather than at the very end.

pelwell reviewed (commented):
There are some fairly horrible bits of code here (fread()ing and malloc()ing most of the file twice?) that I'm not expecting you to fix -  I just have a few comments.

pelwell commented on eepromutils/Makefile:
```diff
@@ -1,14 +1,15 @@
 # Current tools are eepmake and eepdump
 
 CC ?= gcc
+CFLAGS ?= -g -Wall -Wextra
```
And enabling the warnings without fixing the existing warnings isn't very friendly for the average user.

rgetz replied:
Yeah, OK - compiling without warnings turned on is a terrible embedded idea, so I turned them on, and fixed them (as a separate commit)

rgetz commented:
Ok - I updated things, and split things up a bit. I didn't change the format (since there are no spaces anywhere else in the file either, and fixing things with clang-format seemed like a much better idea. (but some don't like this as it makes the history a little harder)

Comments welcome.

rgetz commented on eepromutils/eepdump.c:
```diff
@@ -31,14 +41,25 @@ int read_bin(char *in, char *outf) {
+	fprintf(out, "# --Header--\n# signature=0x%08x (%c%c%c%c)\n# version=0x%02x\n"
+			"# reserved=%u\n# numatoms=%u\n# eeplen=%u\n# ----------\n\n\n",
+			header.signature,
+			(char)(be32toh(header.signature) >> 24) & 0xFF,
```
Ok - I dropped this in v2.

rgetz commented on eepromutils/eeptypes.h:
```diff
@@ -27,6 +27,9 @@
 #define CRC16 0x8005
 
 /* EEPROM header structure */
+//#define HEADER_SIGN 0x69502d52 //"R-Pi" in ASCII reversed for endianness
```
Ok - gone in v2.

rgetz commented on eepromutils/eepdump.c:
```diff
@@ -17,12 +18,21 @@ int read_bin(char *in, char *outf) {
 		printf("Error reading file %s\n", in);
 		return -1;
 	}
+	fseek(fp,0,SEEK_END);
```
overall - formatting is pretty terrible in the whole file - I will run clang-format on the whole things if you don't mind the noise.

pelwell commented:
Thanks - that's better. I'd rather skip the wholesale reformat, though.

rgetz commented:
thanks for merging.
